volatility3/framework/plugins/linux/sockstat.py

```diff
@@ -0,0 +1,506 @@
+            except exceptions.SymbolError as e:
+                # Cannot finds the *_sock type in the symbols
+                vollog.log(constants.LOGLEVEL_V, "Error processing socket family '%s': %s", family, e)
+        else:
+            vollog.log(constants.LOGLEVEL_V, "Unsupported family '%s'", family)
+
+        # Even if the sock family is not supported, or the required types
+        # are not present in the symbols, we can still show some general
+        # information about the socket that may be helpful.
+        saddr_tag = daddr_tag = "?"
+        state = sock.get_state()
+
+        sock_stat = saddr_tag, daddr_tag, state
+
+        return sock, sock_stat, extended
+
+    def _update_extended_socket_filters_info(self, sock: objects.Pointer, extended: dict) -> None:
+        """Get infomation from the socket and reuseport filters
+
+        Args:
+            sock: The kernel sock (sk) struct
+            extended: Dictionary to store extended information
+        """
+        if sock.has_member("sk_filter") and sock.sk_filter:
+            sock_filter = sock.sk_filter
+            extended["filter_type"] = "socket_filter"
+            self._extract_socket_filter_info(sock_filter, extended)
+
+        if sock.has_member("sk_reuseport_cb") and sock.sk_reuseport_cb:
+            sock_reuseport_cb = sock.sk_reuseport_cb
+            extended["filter_type"] = "reuseport_filter"
+            self._extract_socket_filter_info(sock_reuseport_cb, extended)
+
+    def _extract_socket_filter_info(self, sock_filter: objects.Pointer, extended: dict) -> None:
+        extended["bpf_filter_type"] = "cBPF"
+
+        if not sock_filter.has_member("prog") or not sock_filter.prog:
+            return
+
+        bpfprog = sock_filter.prog
+
+        # BPF_PROG_TYPE_UNSPEC = 0
+        if bpfprog.type > 0:
+            extended["bpf_filter_type"] = "eBPF"
+            bpfprog_aux = bpfprog.aux
+            if bpfprog_aux:
+                extended["bpf_filter_id"] = str(bpfprog_aux.id)
+                bpfprog_name = utility.array_to_string(bpfprog_aux.name)
+                if bpfprog_name:
+                    extended["bpf_filter_name"] = bpfprog_name
+
+    def _unix_sock(self, sock: objects.StructType) -> Tuple[objects.StructType, Tuple[str, str, str]]:
+        """Handles the AF_UNIX socket family
+
+        Args:
+            sock: Kernel generic `sock` object
+
+        Returns:
+            unix_sock: The kernel's `unix_sock` object
+            sock_stat: A tuple with the source, destination and state strings.
+        """
+        unix_sock = sock.cast("unix_sock")
+        state = unix_sock.get_state()
+        saddr = unix_sock.get_name()
+        sinode = unix_sock.get_inode()
+        if unix_sock.peer != 0:
+            peer = unix_sock.peer.dereference().cast("unix_sock")
+            daddr = peer.get_name()
+            dinode = peer.get_inode()
+        else:
+            daddr = dinode = ""
```

Comment:
This feels like another one of those places where you may want to make it clear that you couldn't get valid data out?  I'd probably make `daddr_tag` to be `None` then ensure the caller checks the `sock_stat` value for errors before use.